libqpdf/QPDF.cc

```diff
@@ -172,9 +173,8 @@ QPDF::QPDF() :
     // Generate a unique ID. It just has to be unique among all QPDF
     // objects allocated throughout the lifetime of this running
     // application.
-    m->unique_id = static_cast<unsigned long>(QUtil::get_current_time());
-    m->unique_id <<= 32;
-    m->unique_id |= static_cast<unsigned long>(QUtil::random());
+    static std::atomic<unsigned long long> unique_id{0};
```

Comment:
Yes, this looks right to me. I rely on safe static initialization in other parts of the code. It was the one thing I used C++-11 for that I could not have done without it unless I wanted to add dependencies on threading libraries.